test/mongodb.test.js

```diff
@@ -1032,6 +1032,27 @@ describe('mongodb connector', function() {
   });
 
   describe('updateAll', function() {
+    it('should not mutate the input data object', async function() {
+      let user = await User.create({name: 'Al', age: 31, email: 'al@strongloop'});
+      const userId = user.id;
+      user = user.toObject();
+      user.age = 100;
+
+      await User.update(user);
+      user.should.have.property('id', userId);
+    });
+
+    it('should not mutate the input model instance', async function() {
```

Comment:
Maybe I misunderstand, but the test name is saying it _shouldn't_ mutate input model instance, but we do an update which does change the instance? Can you please clarify the use case?

Reply:
@bajtos what are your thoughts? Since you suggested the name.

Reply:
Here is the essence of the test:

```ts
const user = await User.findById(someId);
user.name = 'new name';
await User.update(user);
```

In the test, we are verifying that `user` was not changed by the operation (e.g. the `id` property was not deleted).

Of course, the data stored in the database is modified by the operation.

I proposed to use names "input data object" and "input model instance" to describe content of the variable passed in `User.update` arguments.

@b-admike can you suggest a better naming scheme?

Reply:
Ah okay, makes sense. I get that by `mutate` we mean we don't change the model instance in-place but do a copy instead. I think the name is good as it is, thank you for the explanation.